Sources/MapboxCoreNavigation/RouteController.swift

```diff
@@ -281,59 +281,38 @@ open class RouteController: NSObject {
     private func updateNavigator(with indexedRouteResponse: IndexedRouteResponse,
                                  fromLegIndex legIndex: Int,
                                  completion: ((Result<(RouteInfo?, [AlternativeRoute]), Error>) -> Void)?) {
-        let routeOptions = indexedRouteResponse.validatedRouteOptions
-        
-        let encoder = JSONEncoder()
-        encoder.userInfo[.options] = routeOptions
         guard let newMainRoute = indexedRouteResponse.currentRoute,
-              let routeData = try? encoder.encode(indexedRouteResponse.routeResponse),
-              let routeJSONString = String(data: routeData, encoding: .utf8) else {
-                  completion?(.failure(RouteControllerError.failedToSerializeRoute))
-                  return
+              let routesData = indexedRouteResponse.routesData(routeParserType: routeParserType) else {
+            completion?(.failure(RouteControllerError.failedToSerializeRoute))
+            return
         }
-        
-        let routeRequest = Directions(credentials: indexedRouteResponse.routeResponse.credentials)
-                                .url(forCalculating: routeOptions).absoluteString
-        
-        let parsedRoutes = routeParserType.parseDirectionsResponse(forResponse: routeJSONString,
-                                                                   request: routeRequest,
-                                                                   routeOrigin: indexedRouteResponse.responseOrigin)
-        if parsedRoutes.isValue(),
-           var routes = parsedRoutes.value as? [RouteInterface],
-           routes.count > indexedRouteResponse.routeIndex {
-            self.sharedNavigator.setRoutes(routes.remove(at: indexedRouteResponse.routeIndex),
-                                           uuid: sessionUUID,
-                                           legIndex: UInt32(legIndex),
-                                           alternativeRoutes: routes) { [weak self] result in
-                guard let self = self else { return }
+        self.sharedNavigator.setRoutes(routesData,
+                                       uuid: sessionUUID,
+                                       legIndex: UInt32(legIndex),
+                                       completion:  { [weak self] result in
+            guard let self = self else { return }
+            
+            switch result {
+            case .success(let info):
+                let alternativeRoutes = info.1.compactMap {
+                    AlternativeRoute(mainRoute: newMainRoute,
+                                     alternativeRoute: $0)
+                }
+                completion?(.success((info.0, alternativeRoutes)))
```

Comment:
Here's a question: should we still pass `alternativeRoutes` as one of the callback's parameters? Previously, we filtered and only passed on successfully processed Nav Native `alternativeRoutes`. Now this filtering does not occur.

And also I don't see any place in code where we actually use `alternativeRoutes` parameter in the callback, so we can remove `alternativeRoutes` creation here and not pass it to the callback. But possibly I miss something.

Reply:
What do you mean by saying that filtering is not applied? We are processing navigator output from `info.1` which contains parsed alternative routes as before. 
Right, we are not actually using callback result, but `alternativeRoutes` are still needed as it is reported via `report(newAlternativeRoutes:removedAlternativeRoutes:)` method below.

Reply:
> But possibly I miss something.

I realized that my comment was really not correct, I messed up 2 callbacks, `routesSetupHandler` and `alternativeRoutesSetupHandler`. So everything is actually fine with filtering :)

And you are correct, we use `alternativeRoutes` right in this method so they cannot be deleted.